### PR TITLE
Edit flow for knatified workload for git import with buildConfig, Pipelines and Container Image

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5479

**Solution Description**: 
Edit Knative service for Deployment/Deployment Config created as part of knatified workload
- Edit flow with BuildConfig will share the resource with the original workload and created KSVC
- Edit flow with Pipeline template will not share the resource with original workload and created KSVC, but create one on edit
- Made use of the label `app.kubernetes.io/name` to identify shared resources
- Refactored edit flow to fetch resources from hooks and shared resources are fetched based on label selector for `app.kubernetes.io/name`
- Refactored uploadJar utils and moved EditAction for KSVC to its own package

**Unit test coverage report**: 
Updated specs for

- upload-jar-submit-utils.spec.ts
- knative-modify-applicaton.spec.ts
- kebab-actions.spec.ts
- topology-utils.spec.ts
- modify-application.spec.ts
- edit-application-utils.spec.ts
- knatify-utils.spec.ts 
- pipeline-template-utils.spec.ts 

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/edit-application/EditApplicationPage.tsx [hunk 1]
```diff
@@ -1,99 +1,143 @@
...
+  const [editResData, isEditResDataLoaded, editResDataLoadError] = useK8sWatchResource<
+    K8sResourceKind
+  >(watchedEditResource);
+
+  const watchedResources = React.useMemo(() => {
+    const NAME_LABEL = 'app.kubernetes.io/name';
+    const nameLabel =
+      isEditResDataLoaded &&
+      !editResDataLoadError &&
+      (editResData?.metadata?.labels?.['app.kubernetes.io/name'] || appName);
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/edit-application/EditApplicationPage.tsx:58]:
```suggestion
      (editResData?.metadata?.labels?.[NAME_LABEL] || appName);
```

File: frontend/packages/dev-console/src/components/edit-application/EditApplicationComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  resources: AppResources;
+};
+
+const EditApplicationComponent: React.FunctionComponent<EditApplicationComponentProps> = (
+  props,
+) => {
+  const { appName, resources } = props;
+  const appLabel = resources.editAppResource?.data?.metadata?.labels?.['app.kubernetes.io/name'];
+
+  const getAssociatedResouce = (resourcesObj: WatchK8sResultsObject<K8sResourceKind[]>) => {
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/edit-application/EditApplicationComponent.tsx:29]:
```suggestion
  const getAssociatedResource = (resourcesObj: WatchK8sResultsObject<K8sResourceKind[]>) => {
```

File: frontend/packages/dev-console/src/components/edit-application/EditApplicationPage.tsx [hunk 3]
```diff
@@ -1,99 +1,143 @@
...
-import { EditApplicationProps } from './edit-application-types';
+import EditApplicationComponent from './EditApplicationComponent';
 
-const INSTANCE_LABEL = 'app.kubernetes.io/instance';
-const EditApplicationComponentLoader: React.FunctionComponent<EditApplicationProps> = (
-  props: EditApplicationProps,
-) => {
-  const { loaded } = props;
-  return loaded ? <EditApplication {...props} /> : <LoadingBox />;
+type watchResource = {
```
Comment (karthikjeeyar) [inline frontend/packages/dev-console/src/components/edit-application/EditApplicationPage.tsx:19]: type should be in pascal case
```suggestion
type WatchResource = {
```

**Comment by karthikjeeyar** [comment]
Verified the scenarios with GIT Import, Dockerfile, along with health checks, HPA, env's and other advanced sections properties. It works as expected.
/lgtm

**Review by sahil143 — APPROVED** [review]
/lgtm

verified locally, works as expected.

**Comment by christianvogt** [comment]
/approve

please rebase

**Comment by invincibleJai** [comment]
/close 

opened a new PR https://github.com/openshift/console/pull/8593 as label `needs-rebase` not getting updated :/

**Pull request closed without merging**